api/encoding/outbound_call.go

```diff
@@ -75,6 +75,30 @@ func (c *OutboundCall) WriteToRequest(ctx context.Context, req *transport.Reques
 	return ctx, nil
 }
 
+// WriteToRequestMeta fills the given request with request-specific options from
+// the call.
+//
+// The context MAY be replaced by the OutboundCall.
+func (c *OutboundCall) WriteToRequestMeta(ctx context.Context, reqMeta *transport.RequestMeta) (context.Context, error) {
+	for _, h := range c.headers {
+		reqMeta.Headers = reqMeta.Headers.With(h.k, h.v)
```

Comment:
Optional: should we change WriteToRequest to,

    return c.WriteToRequestMeta(ctx, req.ToMeta())

Reply:
I think we did this so we don't make a copyh

Reply:
I just realized my comment here was incorrect since you're writing *to* the meta, not reading from it.